:sparkles:  Resolution page v2

rszwajko opened the pull request:
Depends on: #195 

Changes:
1. show Resolutions page from Issue tree view via conext button
2. track solution state and collect log messages
3. use chat-like layout
4. support Markdown
5. support light/dark mode

rszwajko commented:
## Resolution page in light mode
![Screenshot from 2025-01-16 23-54-15](https://github.com/user-attachments/assets/8583fff9-6f6d-4f27-b859-11caf7e654b5)
![Screenshot from 2025-01-17 00-04-27](https://github.com/user-attachments/assets/1e4c1618-fe5e-49f4-8295-aa60671b554b)

## Analysis page in light mode with markdown
![Screenshot from 2025-01-16 23-53-38](https://github.com/user-attachments/assets/3ef484a0-403d-4daf-90ae-6d0169e2a219)

## Resolution page in dark mode
![Screenshot from 2025-01-17 00-08-05](https://github.com/user-attachments/assets/15806d53-53ad-44f0-a878-6c6d1560b6c8)

## Opening Resolution page from Issue tree view
![Screenshot from 2025-01-17 00-08-34](https://github.com/user-attachments/assets/f0f6e737-8f6f-4225-9901-a1793f7828a9)

rszwajko commented:
## Updated styles for file changes

![Screenshot from 2025-01-17 15-14-33](https://github.com/user-attachments/assets/b7a6a3a9-22f6-49c9-8b82-a429adf350d1)
![Screenshot from 2025-01-17 15-14-06](https://github.com/user-attachments/assets/8c046477-72d6-4026-a212-73c2cf407c3f)

djzager reviewed (commented):
This is really awesome to see! Nice work.

The only issue I see is in Dark Mode, the white text on blue label is hard to read. I'm not sure if the right way to fix it is to change what "blue" is in dark mode, or to change the color on the label when in dark mode.

rszwajko commented:
@djzager 
> The only issue I see is in Dark Mode, the white text on blue label is hard to read
>
yep.  check the latest commit - I've switched to black.

ibolton336 commented:
Noticing an issue with styling. 

<img width="1552" alt="Screenshot 2025-01-20 at 12 50 18 PM" src="https://github.com/user-attachments/assets/9a74bc1d-fba5-48c1-b616-a000656ed326" />

The page component is no longer center aligned and a large space to left is visible when viewing analysis results.

rszwajko commented:
@sjdy78 @ibolton336 
The layout problems seems related to missing sidebar - after providing a dummy hidden sidebar it looks OK without any tweaks.
Additionally in the latest commit:
1.  I've fixed double scrollbars by moving "Run analysis" button to the masthead  and removing manual height calculation.
2. Removed "Cancel" button as suggested in #178

rszwajko commented:
### Scrollbars and buttons in the masthead
![Screenshot from 2025-01-20 22-59-50](https://github.com/user-attachments/assets/291aebe4-b936-48be-a9aa-9f15929fb7e8)
![Screenshot from 2025-01-20 22-55-06](https://github.com/user-attachments/assets/23d51457-984e-4229-9507-f4bc96d06563)

rszwajko commented:
Force pushed to rebase on top of current HEAD.

sjd78 commented on webview-ui/src/components/ResolutionsPage.tsx:
```diff
@@ -1,169 +1,209 @@
-      </Page>
-    );
-  }
+  console.log("Resolution view state:", {
```
useful to keep this log?

rszwajko replied:
Yes. Unfortunately the logic got complex in order to provide chat-like experience. The extra flags may be hard to debug - especially that LLM response/errors might not be easy to reproduce.

sjd78 reviewed (approved):
The screen shots look pretty good.  I expect some followup tweaking as it gets more use, but it is a solid base.  Code structure is reasonable.  I only found one lingering console.log that could be kept if it makes sense.

Fighting to get the vscode themes to line up with PF6 styles will be ongoing.  Looks a good at least 90% of the way there and that feels good enough at this point.
